pallets/grant/src/lib.rs

```diff
@@ -172,43 +191,47 @@ pub mod pallet {
 	impl<T: Config> Pallet<T> {
 
 		/// Dispatchable call that ensures grants can be requested
-		#[pallet::weight(<T as Config>::WeightInfo::request_grant())]
-		pub fn request_grant(origin: OriginFor<T>, grant_requester: T::AccountId) -> DispatchResult {
+		#[pallet::weight((<T as Config>::WeightInfo::request_grant(), Pays::No))]
+		pub fn request_grant(origin: OriginFor<T>) -> DispatchResult {
 
 			// Check that the extrinsic was signed and get the signer.
-			let _account = ensure_signed(origin)?;
+			let account = ensure_signed(origin)?;
+
+			ensure!(Self::storage_requesters(&account).is_none(), Error::<T>::RequestAlreadyMade);
+
+			ensure!(T::Currency::free_balance(&account) == T::ExistentialDeposit::get(), Error::<T>::NonEmptyBalance);
```

Comment:
I am getting nonEmptyBalance error even when balance is empty. There is a bug somewhere?

Reply:
Balance cannot be empty, balance must be the initial balance set by the system (Existential Deposit).
Balances below this amount are dropped from storage.

Reply:
Only completely new wallets can interact with the grant pallet.